sympy/polys/series/ring.py

```diff
@@ -1,109 +1,1100 @@
+        >>> from sympy import QQ
+        >>> from sympy.polys.series import PowerSeriesRingField
+        >>> R = PowerSeriesRingField(QQ, "x")
+        >>> x = R.gen
+        >>> R.cos(x)
+        1 - 1/2*x**2 + 1/24*x**4 + O(x**6)
+
+        """
+        R = self.ring
+        series = R.cos(s.series)
+        return self.from_element(series)
+
+    def cosh(self, s: PowerSeriesElement[Ef]) -> PowerSeriesElement[Ef]:
+        """
+        Compute the hyperbolic cosine of a power series.
+
+        The constant term should be zero for proper expansion in the Rational Field.
+
+        Examples
+        ========
+        >>> from sympy import QQ
+        >>> from sympy.polys.series import PowerSeriesRingField
+        >>> R = PowerSeriesRingField(QQ, "x")
+        >>> x = R.gen
+        >>> R.cosh(x)
+        1 + 1/2*x**2 + 1/24*x**4 + O(x**6)
+
+        """
+        R = self.ring
+        series = R.cosh(s.series)
+        return self.from_element(series)
+
+
+class PowerSeriesElement(DomainElement, CantSympify, Generic[Er]):
+    """A class for representing elements of a Power Series."""
+
+    def __init__(self, ring: PowerSeriesRingRing[Er], series: TSeriesElement[Er]):
+        self.series = series
+        self._parent_ring = ring
+        self.ring = ring.ring
+        self.domain = ring.domain
+        self.symbol = ring.symbol
+
+    def __repr__(self) -> str:
+        R = self.ring
+        return R.pretty(self.series, symbol=str(self.symbol))
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, PowerSeriesElement):
+            return False
+
+        R = self.ring
+        return R.equal_repr(self.series, other.series)
+
+    def __hash__(self) -> int:
+        return hash((self._parent_ring, self.series))
+
+    def __pos__(self) -> PowerSeriesElement[Er]:
+        R = self.ring
+        s = R.positive(self.series)
+        return self.new(s)
+
+    def __neg__(self) -> PowerSeriesElement[Er]:
+        R = self.ring
+        s = R.negative(self.series)
+        return self.new(s)
+
+    def new(self, series: TSeriesElement[Er]) -> PowerSeriesElement[Er]:
+        return self.__class__(self._parent_ring, series)
+
+    def __add__(
+        self, other: PowerSeriesElement[Er] | Er | int
+    ) -> PowerSeriesElement[Er]:
+        """
+        Add two power Series.
+
+        Examples
+        ========
+
+        >>> from sympy import QQ
+        >>> from sympy.polys.series import PowerSeriesRingField
+        >>> R = PowerSeriesRingField(QQ, "x")
+        >>> x = R.gen
+        >>> R.sin(x) + R.cos(x)
+        1 + x - 1/2*x**2 - 1/6*x**3 + 1/24*x**4 + 1/120*x**5 + O(x**6)
+
+        """
+        if not other:
+            if isinstance(other, (PowerSeriesElement, int)) or self.domain.of_type(
+                other
+            ):
+                return self
+            raise TypeError(
+                f"Unsupported type {type(other).__name__} for addition with PowerSeriesElement."
+            )
```

Comment:
This isn't the right way to handle this. It is better to remove the `if not other` check here at the start of the function. The types should be checked first (as they are below). Once we know we have the correct type we can check `if not other` if that is worth doing. Possibly the right place for this is in `_add` or `_add_ground` rather than here where we should really just be dispatching on types.